voxblox/include/voxblox/integrator/tsdf_integrator.h

```diff
@@ -75,16 +73,18 @@ class TsdfIntegratorBase {
   // Will return a pointer to a voxel located at global_voxel_idx in the tsdf
   // layer. Thread safe.
   // Takes in the last_block_idx and last_block to prevent unneeded map lookups.
-  // If the block this voxel would be in has not been allocated, a voxel in
-  // temp_voxel_storage is allocated and returned instead.
-  // This can be merged into the layer later by calling
-  // updateLayerWithStoredVoxels(temp_voxel_storage)
-  inline TsdfVoxel* findOrTempAllocateVoxelPtr(
-      const VoxelIndex& global_voxel_idx, Block<TsdfVoxel>::Ptr* last_block,
-      BlockIndex* last_block_idx, VoxelMap* temp_voxel_storage) const;
+  // If the block this voxel would be in has not been allocated, a block in
+  // temp_block_map_ is created/accessed and a voxel from this map is returned
+  // instead. Unlike the layer, accessing temp_block_map_ is controlled via a
+  // mutex allowing it to grow during integration.
+  // These temporary blocks can be merged into the layer later by calling
+  // updateLayerWithStoredBlocks()
+  inline TsdfVoxel* getVoxelPtr(const VoxelIndex& global_voxel_idx,
```

Comment:
could you change the name to sth more descriptive, like before? getVoxelPtr sounds like any other getter even though there is some black magic going on.